### PR TITLE
add explicit flag for externally provisioned hosts

### PR DESCRIPTION
**Pull request by dhellmann**
We want to add an externallyProvisioned flag to the host specification to make it easier to keep track of hosts that have already had an image written to them, so we don't overwrite it.

Addresses #238

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -30,6 +30,12 @@ power state on the physical host.
 host. For example, a Machine when the host is being used by the
 machine-api.
 
+*externallyProvisioned* -- A boolean indicating whether something else
+is managing the image running on the host. When set, if no image is
+provided, the host's power status and hardware inventory will be
+monitored. If the *image* contains image parameters, this field is
+ignored.
```
Comment (russellb) [inline docs/api.md]: why not just always ignore the image field if externallyProvisioned is true?

Comment (dhellmann) [reply docs/api.md]: The user can change the externallyProvisioned field at any time. So, they could start to provision a host in the normal way and then while it's in the middle of provisioning set externallyProvisioned=true. At that point, the logic managing the provisioning workflow would be circumvented and the host status would never update properly.

We _could_ allow that, but it's easier to reason about the host if we say that giving an image and saying a host is externally provisioned are mutually exclusive, but the image has precedence if both are provided. We end up with fewer overall transitions in the state diagram that way.

Comment (russellb) [reply docs/api.md]: OK, that makes sense, thanks.

Comment (dhellmann) [reply docs/api.md]: I added some comments in the code to explain the reasoning for this choice.

**Comment by dhellmann** [comment]
I tested this locally with the current CAPB by scaling down the worker MachineSet to delete the machine and it worked. Then I scaled the MachineSet back up, provisioned a worker, and then deleted the Machine directly. The worker was deprovisioned and then reprovisioned by the new Machine created by the MachineSet.

**Pull request merged**